File: problem06/p06.js
function createNumbers() {
  var numbers = [];
  for (var i=0; i <= 100; i++) {
    numbers.push(i);
  }
  return numbers;
}

function sqNumberSum(numbers) {
  var total = numbers.reduce(function(a,b) {
    b = Math.pow(b,2);
    return a + b;
  });
  return total;
}

function sumSq(numbers) {
  var total = numbers.reduce(function(a,b) {
    return a + b;
  });
  return Math.pow(total, 2);
}

function solution() {
  var numbers = createNumbers();

  var n1 = sqNumberSum(numbers);
  var n2 = sumSq(numbers);

  return n2 - n1;
}

console.log(solution());

File: problem06/p06.py
import math #import math module

def solution():
    sumOfSq = reduce(lambda x,y: x+math.pow(y,2), range(1,101))
    sqOfSum = math.pow((reduce(lambda x,y: x+y, range(1,101))), 2)

    return sqOfSum - sumOfSq

print solution()
